stripe/src/test/java/com/stripe/android/StripeTest.java

```diff
@@ -849,23 +849,36 @@ public void retrieveSourceAsync_withValidData_passesIntegrationTest() throws Str
         );
     }
 
-    @Test
-    public void retrieveSourceSynchronous_withValidData_passesIntegrationTest()
-            throws StripeException {
-        final Source source = createSource();
-
-        final String sourceId = source.getId();
-        final String clientSecret = source.getClientSecret();
-        assertNotNull(sourceId);
-        assertNotNull(clientSecret);
-
-        final Source retrievedSource = createStripe()
-                .retrieveSourceSynchronous(sourceId, clientSecret);
-
-        // We aren't actually updating the source on the server, so the two sources should
-        // be identical.
-        assertEquals(source, retrievedSource);
-    }
+  @Test
+  public void retrieveSourceSynchronous_withValidData_passesIntegrationTest()
+      throws StripeException {
+    final Source source = createSource();
+
+    final String sourceId = source.getId();
+    final String clientSecret = source.getClientSecret();
+    assertNotNull(sourceId);
+    assertNotNull(clientSecret);
+
+    final Source retrievedSource =
+        createStripe(ApiKeyFixtures.DEFAULT_PUBLISHABLE_KEY, fireAndForgetRequestExecutor)
+            .retrieveSourceSynchronous(sourceId, clientSecret);
+
+    // We aren't actually updating the source on the server, so the two sources should
+    // be identical.
+    assertEquals(source, retrievedSource);
+
+    verify(fireAndForgetRequestExecutor).executeAsync(stripeRequestArgumentCaptor.capture());
+    final StripeRequest analyticsRequest = stripeRequestArgumentCaptor.getValue();
+    assertEquals(AnalyticsRequestFactory.HOST, analyticsRequest.getBaseUrl());
+
+    assertEquals(
+        AnalyticsEvent.SourceRetrieve.toString(),
+        Objects.requireNonNull(analyticsRequest.getParams()).get(AnalyticsDataFactory.FIELD_EVENT));
+    assertEquals(
```

Comment:
I've started moving towards using Google's [truth](https://truth.dev/) assertions library. It's similar to assertJ, so you can do

```
    assertThat(actual)
        .isEqualTo(expected)
```